sdk/core/azure-core/inc/azure/core/http/curl/curl_session.hpp

```diff
@@ -319,14 +319,25 @@ namespace Azure { namespace Core { namespace Http {
       return eof && m_sessionState != SessionState::PERFORM;
     }
 
+    /**
+     * @brief All connections will request to keep the channel open to re-use the
+     * connection.
+     *
+     * @remark This option can be disabled from the transport adapter options. When disabled, the
+     * session won't return connections to the connection pool. Connection will be closed as soon as
+     * the request is completed.
+     *
+     */
+    bool m_keepAlive = true;
+
   public:
     /**
      * @brief Construct a new Curl Session object. Init internal libcurl handler.
      *
      * @param request reference to an HTTP Request.
      */
-    CurlSession(Request& request, std::unique_ptr<CurlNetworkConnection> connection)
-        : m_connection(std::move(connection)), m_request(request)
+    CurlSession(Request& request, std::unique_ptr<CurlNetworkConnection> connection, bool keepAlive)
```

Comment:
My preference is not to use bools.  Use an enum instead, Makes the intention of the caller very clear.
enum class CurlSessionHandleBehavior {
   NoReuse,
   KeepAlive
};
CurlSession(Request& request, std::unique_ptr<CurlNetworkConnection> connection, CurlSessionHandleBehavior 
 keepAlive)

Reply:
That would be an extra type to be exposed to the surface. Do we want that?
KeepAlive is a setting from the `CurlTransportOptions`, it would be the only setting with the enum class (all other settings are bools).

Because if I only let the session with an enum class, then I would need to check the bool setting to chose the enum value, like
```cpp
CurlSessionHandleBehavior option = CurlSessionHandleBehavior::KeepAlive; //default
if (!curlTransportOptions.keepAlive)
   option = CurlSessionHandleBehavior::NoReuse;

 CurlSession(request, connection, option)
```